### DISCUSSION TITLE
Functional inconsistency discovered, while changing from HACS to Core integration

### DISCUSSION THREAD

**Discussion by Lateadapter** [music-assistant/support#3289] [Q&A] [answered]
Within some automation I'm using tts anouncement for Sonos-speakers. Due to the move to MA core integration I was forced to change all previous used entity names. Obviously I changed too much, because afterwards some strange behaviour occured.
However, finally I solved it for me but actual I found some functional inconsistency between MASS- and HA-player entities.

I'm using the core integration from Sonos as well as now the core integration from MASS with activated Sonos speakers.
Therefore I have now double entities for same speakers - in following sample either "media_player.ha_sonos_kuche" (from Sonos core integration) resp. "media_player.ma_sonos_kuche" (from MASS). But the behavior for same player service is totally different!

- Following code behaves as wanted:  I get a tts message with defined volume, but without announcement jingle
- Sonos entity, "announce: true" (parameter name questionable however behavior as documented) --> no jingle, volume as defined

![image](https://github.com/user-attachments/assets/ef8a25f5-1f50-49eb-a2d7-ccc154877e1e)

- Sonos entity, "announce: false" --> no jingle, volume definition ignored - current volume adjustment used

![image](https://github.com/user-attachments/assets/623eb423-48a1-4bc7-949e-10c702f42577)

- MASS entity, "announce: true" --> jingle activ, volume definition ignored but actual volume very high

![image](https://github.com/user-attachments/assets/090cc4da-0b21-43e0-9273-30574e986d73)

- MASS entity, "announce: false" --> no jingle, volume definition ignored but actual volume very low

![image](https://github.com/user-attachments/assets/a9a1c87d-a8cd-4620-8189-02984569094e)

Same YAML-code, same speaker but completely different behavior. What I need, will be done by first sample.
I would not say that the Sonos-core is doing a perfect logic thing, however it behaves as documented for long time.
Thank you very much!

For the implementation of future functional extensions in MASS, I would recommend also examining existing solutions and adopting them if necessary.
MASS made very good progress within recent month and I like it very much. My hope is that one day any audial functionality can be moved to that integration.

**Comment by OzGav** [comment, marked as answer]
You are not using the correct action nor the correct uri for the MA players.

See:
https://music-assistant.io/integration/announcements/
https://music-assistant.io/faq/how-to/#get-the-uri
